modules/offlinemaps/src/main/java/com/sygic/maps/offlinemaps/loader/MapLoaderGlobal.kt

```diff
@@ -0,0 +1,271 @@
+/*
+ * Copyright (c) 2020 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.offlinemaps.loader
+
+import androidx.lifecycle.MutableLiveData
+import com.sygic.maps.uikit.views.common.livedata.SingleLiveEvent
+import com.sygic.maps.uikit.views.common.utils.logError
+import com.sygic.maps.uikit.views.common.utils.logWarning
+import com.sygic.sdk.map.MapLoader
+import com.sygic.sdk.map.listeners.MapInstallProgressListener
+import com.sygic.sdk.map.listeners.MapResumedInstallDoneListener
+import kotlinx.coroutines.*
+import java.util.concurrent.Executors
+import kotlin.coroutines.CoroutineContext
+import kotlin.coroutines.EmptyCoroutineContext
+
+object MapLoaderGlobal {
```

Comment:
Skod by sa mi pacilo ak by to bol injectovany singleton za interfaceom, Potom sa to da mockovat a testovat. Ak to potrebujes ako object tak ok